prompting/datasets/base.py

```diff
@@ -0,0 +1,61 @@
+import time
+from abc import ABC, abstractmethod
+from typing import Dict, Literal
+from pydantic import BaseModel
+from typing import ClassVar
+
+
+class Context(BaseModel):
+    # TODO: Pydantic model
+    title: str
+    topic: str
+    subtopic: str
+    content: str
+    internal_links: list[str]
```

Comment:
I said this elsewhere but I'll say it here because you literally have `from typing import Dict` :) you should add `List` to the import and then here use `List[str]`